### PR TITLE
Behavior initial state sampling

### LINKED ISSUE #243: BEHAVIOR: Incorporate an `rng` into the placeOnTop Sampler
The `placeOnTop` sampler internally doesn't use an rng, which means we can't control it's randomness. Fix this.

### PR DESCRIPTION
**Pull request by NishanthJKumar**
Enables us to sample new initial states for scenes in a seeded fashion so that we can run our pipeline on multiple instances of a task!
Also seeds the sampler for the placeOnTop option and thereby resolves  #243

### CODE REVIEW

File: src/envs/behavior.py [hunk 1]
```diff
@@ -164,12 +167,20 @@ def _get_tasks(self, num: int, rng: np.random.Generator) -> List[Task]:
         for _ in range(num):
             # Behavior uses np.random everywhere. This is a somewhat
             # hacky workaround for that.
-            np.random.seed(rng.integers(0, (2**32) - 1))
-            self.igibson_behavior_env.reset()
+            curr_env_seed = rng.integers(0, (2**32) - 1)
+            if CFG.behavior_randomize_init_state:
+                self.set_igibson_behavior_env(curr_env_seed)
+                self.igibson_behavior_env.reset()
```
Comment (ronuchit) [inline src/envs/behavior.py:173]: Can you pull this line out of the `if` and delete the else?

File: src/envs/behavior.py [hunk 2]
```diff
@@ -301,6 +312,44 @@ def render(self,
...
+                    physics_timestep=CFG.behavior_physics_timestep,
+                    action_filter="mobile_manipulation",
+                    rng=self._rng,
+                )
+                self.igibson_behavior_env.step(
+                    np.zeros(self.igibson_behavior_env.action_space.shape))
+                ig_objs_bddl_scope = [
+                    self._ig_object_name(obj)
+                    for obj in self._get_task_relevant_objects()
+                ]
```
Comment (ronuchit) [inline src/envs/behavior.py:351]: No need to duplicate the code in this loop with the code above. A better pattern is:
```
while True:
    self.igibson_behavior_env = blah
    self.igibson_behavior_env.step()
    ig_objs_bddl_scope = [blah]
    if None not in ig_objs_bddl_scope:
        break
```

Comment (ronuchit) [reply src/envs/behavior.py:351]: One concern is, how likely is it that this loop might run forever? Should we impose a max number of iterations before killing it or something?

**Review by ronuchit — COMMENTED** [review]
aside from my comments, rest lgtm, but let @tomsilver look too

File: src/envs/behavior.py (hunk 2 above)
Comment (NishanthJKumar) [reply src/envs/behavior.py:351]: from our tests, there's a very low probability this will run forever, but I'll add a counter just in case!

**Review by tomsilver — APPROVED** [review]
great job!

**Pull request merged**